From 6590057dd2d347f3a0ab8095df4a72abce1fb85d Mon Sep 17 00:00:00 2001
From: WangJing <wangjing@mprc.pku.edu.cn>
Date: Mon, 10 Oct 2011 17:13:59 +0800
Subject: [PATCH 008/641] UniCore64: Add definition of KSTK_ESP and KSTK_EIP

Fixed the error
  fs/proc/array.c:388:4: error: implicit declaration of function 'KSTK_EIP'
  fs/proc/array.c:389:4: error: implicit declaration of function 'KSTK_ESP'

Add definition of KSTK_ESP and KSTK_EIP in processor.h

Signed-off-by: WangJing <wangjing@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/processor.h |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/unicore64/include/asm/processor.h b/arch/unicore64/include/asm/processor.h
index d071a3f..0506482 100644
--- a/arch/unicore64/include/asm/processor.h
+++ b/arch/unicore64/include/asm/processor.h
@@ -33,6 +33,14 @@ extern int kernel_thread(int (*fn)(void *), void *arg, unsigned long flags);
 
 extern unsigned long get_wchan(struct task_struct *p);
 
+/* Return some info about the user process TASK. */
+#define KSTK_PTREGS_GAP  8 /* FIXME */
+#define task_pt_regs(p)	((struct pt_regs *) \
+		(task_stack_page(p) + THREAD_SIZE - KSTK_PTREGS_GAP) - 1)
+/* Aliases for pc and sp (used in fs/proc/array.c) */
+#define KSTK_EIP(tsk)  (task_pt_regs(tsk)->UC64_R31)
+#define KSTK_ESP(tsk)  (task_pt_regs(tsk)->UC64_R29)
+
 #endif /* __KERNEL__ */
 
 #endif /* __ASM_UNICORE64_PROCESSOR_H__ */
-- 
1.7.9.5

